pkg/agent/cluster.go

```diff
@@ -0,0 +1,330 @@
+		RestAPIPreviousClusterStatus:                        "",
+		RestAPIHostValidationsPassed:                        false,
+		ClusterKubeAPISeen:                                  false,
+	}
+
+	czero.Ctx = ctx
+	czero.API = capi
+	czero.clusterID = clusterID
+	czero.clusterInfraEnvID = clusterInfraEnvID
+	czero.installHistory = cinstallstatushistory
+	return czero, nil
+}
+
+// Get Retrieve the current cluster metadata from the Agent Rest API
+func (czero *Cluster) Get() (*models.Cluster, error) {
+	// GET /v2/clusters/{cluster_zero_id}
+	getClusterParams := &installer.V2GetClusterParams{ClusterID: *czero.clusterID}
+	result, err := czero.API.Rest.Client.Installer.V2GetCluster(czero.Ctx, getClusterParams)
+	if err != nil {
+		return nil, err
+	}
+	cluster := result.Payload
+	return cluster, nil
+}
+
+// IsBootstrapComplete Determine if the cluster that agent installer
+// is installing has completed the bootstrap process.
+func (czero *Cluster) IsBootstrapComplete() (bool, error) {
+
+	agentRestAPILive, agentRestAPIErr := czero.API.Rest.IsRestAPILive()
+	if agentRestAPIErr != nil {
+		logrus.Debug("node zero Agent API is not available")
+		logrus.Debug(agentRestAPIErr)
+	}
+
+	clusterKubeAPILive, clusterKubeAPIErr := czero.API.Kube.IsKubeAPILive()
+	if clusterKubeAPIErr != nil {
+		logrus.Debug("cluster Kube API is not available")
+		logrus.Debug(clusterKubeAPIErr)
+	}
+
+	if clusterKubeAPILive {
+		// First time we see the cluster Kube API
+		if !czero.installHistory.ClusterKubeAPISeen {
+			logrus.Info("cluster Kube API Initialized")
+			czero.installHistory.ClusterKubeAPISeen = true
+		}
+
+		configmap, _ := czero.API.Kube.IsBootstrapConfigMapComplete()
+		if configmap {
+			logrus.Info("bootstrap configMap status is complete")
+			return true, nil
+		}
+	}
+
+	if agentRestAPILive {
+		// First time we see the agent Rest API
+		if !czero.installHistory.RestAPISeen {
+			logrus.Info("node zero Agent API Initialized")
+			czero.installHistory.RestAPISeen = true
+		}
+		logrus.Trace("getting cluster info from Node Zero Agent API")
+		clusterState, _ := czero.Get()
+
+		// TODO(lranjbar)[AGENT-172]: Add CheckHostValidations
+		// if !czero.installHistory.AgentHostValidationsPassed {
+		// 	validations := CheckHostValidations()
+		// 	if validations {
+		// 		czero.installHistory.AgentHostValidationsPassed = true
+		// 	}
+		// }
+
+		czero.PrintInstallStatus(clusterState)
+		czero.installHistory.RestAPICurrentClusterStatus = *clusterState.Status
+
+		// Update Install History object when we see these states
+		czero.updateInstallHistoryClusterStatus(clusterState)
+
+		stopped, _ := czero.HasStoppedInstalling(*clusterState.Status)
+		if stopped {
+			logrus.Error("cluster has stopped installing")
+			errored, _ := czero.HasErrored(*clusterState.Status)
+			if errored {
+				return false, errors.New("cluster installation has stopped due to errors")
+			}
+			return false, errors.New("cluster has stopped installing and/or insallation was cancelled")
```

Comment:
I don't remember, but do we support (or plan to) the possibility to cancel the installation in the automated workflow?

```suggestion
			return false, errors.New("cluster has stopped installing and/or installation was cancelled")
```

Reply:
I don't know if its planned but someone could do it using the API on node zero before bootstrap reboots.